language/tools/diem-resource-viewer/src/lib.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{bail, Result};
+use diem_types::{
+    access_path::AccessPath, account_address::AccountAddress, account_state::AccountState,
+    contract_event::ContractEvent,
+};
+use move_core_types::language_storage::StructTag;
+use move_vm_runtime::data_cache::MoveStorage;
+use resource_viewer::MoveValueAnnotator;
+use std::{
+    collections::BTreeMap,
+    fmt::{Display, Formatter},
+};
+
+pub use resource_viewer::{AnnotatedMoveStruct, AnnotatedMoveValue};
+
+pub struct DiemValueAnnotator<'a> {
```

Comment:
FWIW, a short comment here explaining why this wrapper exists would have helped me understand this code sooner.

Also, I'm no Rust expert, but: is this a good opportunity to use a single-element tuple struct, i.e.: `pub struct DiemValueAnnotator<'a>(MoveValueAnnotator<'a>);`? I believe it would mostly be the same thing, but `.inner` would be replaced with `.0`. Would there be any benefit aside from typing ~30 fewer characters here?

Reply:
I'll add the comment. Thanks for pointing this out!

Yeah I would say the choice between named and unnamed fields can be quite arbitrary and may depend on taste, or even mood at times. In this case though, I do agree that `inner` is quite meaningless and `.0` seems to be more succinct.